### ISSUE TITLE
Misplaced rendering during scale-up-animation preceded by a transform-animation

### ISSUE THREAD

**Issue by govis** [PolymerElements/neon-animation#221]
### Description

A set of 4 cards is being animated upon page load with <code>transform-animation from translateY(100%) to none</code> (a la neon-animation load demo). After that the cards start refreshing one-by-one with new content with <code>scale-up-animation</code> animation. When search criteria changes the process restarts with all card load and animation. At some point we start seeing card images rendered in the upper-left corner of the page (and outside of the actual card bounds).

### Expected outcome

![image](https://cloud.githubusercontent.com/assets/971651/22529905/52ce41de-e8a7-11e6-9e08-9cc144614d40.png)

### Actual outcome

![image](https://cloud.githubusercontent.com/assets/971651/22529886/335d4eb2-e8a7-11e6-8d19-14a6255cf592.png)

### Live Demo

https://jsbin.com/tanozaqoyo/edit?html,output

### Steps to reproduce

Run the sample, 4 cards should load and start spinning one-by-one. You should start seeing an image poping up in the top-left corner of the page. Click the "Load More" mid-cycle until you see the issue.

### Browsers Affected

- [x] Chrome
- [ ] Firefox
- [ ] Safari 9
- [ ] Safari 8
- [ ] Safari 7
- [ ] Edge
- [ ] IE 11
- [ ] IE 10

**Comment by govis** [comment]
Code to reproduce 

https://jsbin.com/tanozaqoyo/edit?html,output

<pre><code>&lt;!DOCTYPE html>
&lt;html>
&lt;head>
  &lt;meta charset="utf-8">
  &lt;title>Polymer Element Test Case&lt;/title>
  &lt;base href="//polygit.org/components/">
  &lt;script src="webcomponentsjs/webcomponents-lite.js">&lt;/script>
  &lt;link rel="import" href="polymer/polymer.html">
  &lt;!-- Import any relevant elements here: -->
  &lt;!-- Example:
  &lt;link rel="import" href="paper-button/paper-button.html">
  -->
&lt;link rel="import" href="polymer/polymer.html">
&lt;link rel="import" href="iron-flex-layout/iron-flex-layout.html">
&lt;link rel="import" href="iron-flex-layout/iron-flex-layout-classes.html">
&lt;link rel="import" href="paper-button/paper-button.html">
&lt;link rel="import" href="iron-ajax/iron-ajax.html">
&lt;link rel="import" href="iron-image/iron-image.html">
&lt;link rel="import" href="neon-animation/neon-animation-runner-behavior.html">
&lt;link rel="import" href="neon-animation/neon-animations.html">  
  
&lt;dom-module id="main-page">
    &lt;template>
        &lt;style>
            :host {
                @apply(--layout);
                @apply(--layout-fit);
                overflow-y: auto;
            }

            #content {
                width: 100%;
                margin-top: auto;
                margin-bottom: auto;
            }

            .title {
                position: relative;
            }

            paper-button {
                 position: absolute;
                 left: 50%;
                 transform: translateX(-50%);
            }

            card-list {
                display: block;
                margin-top: 40px;
            }
        &lt;/style>

        &lt;div id="content">
            &lt;div class="title">
                &lt;paper-button on-tap="loadPhotos">Load More&lt;/paper-button>
            &lt;/div>
            &lt;card-list id="cards">&lt;/card-list>
        &lt;/div>
    &lt;/template>
    &lt;script>
        Polymer({
            is: 'main-page',

            loadPhotos: function () {
                this.$.cards._loadMorePhotos();
            }
        });
    &lt;/script>
&lt;/dom-module>

&lt;dom-module id="image-card">
    &lt;template>
        &lt;style>
            :host {
                @apply(--layout-vertical);
                @apply(--layout-relative);
                @apply(--shadow-elevation-3dp);
                background-color: var(--card-background, #fff);
                border-radius: 3px;
                position: relative;
                box-sizing: border-box;
                width: 200px;
                height: 300px;
            }

            .head {
                @apply(--layout-relative);
                height: 150px;
            }

            .head iron-image {
                position: absolute;
                top: 50%;
                left: 50%;
                transform: translateY(-50%) translateX(-50%);
            }

            .body {
                @apply(--layout-vertical);
                @apply(--layout-flex);
                @apply(--paper-font-common-base);
                height: 150px;
                font-size: 16px;
                line-height: 20px;
                text-align: left;
                padding: 0 12px 12px;
                color: rgba(0,0,0,.87);
            }

            .name {
                max-height: 40px;
                padding-top: 4px;
                overflow: hidden;
                text-overflow: ellipsis;
                word-break: break-word;
            }

            .text {
                @apply(--layout-flex);
                font-size: 13px;
                font-weight: 300;
                padding-top: 8px;
                overflow: hidden;
                text-overflow: ellipsis;
                color: rgba(0,0,0,.54);
            }
        &lt;/style>
        &lt;div class="head">
            &lt;iron-image style="width:200px; height:150px;" sizing="contain"
                        src="//farm[[photo.farm]].staticflickr.com/[[photo.server]]/[[photo.id]]_[[photo.secret]]_n.jpg">
            &lt;/iron-image>
        &lt;/div>
        &lt;div class="body">
            &lt;div class="name">
                &lt;span>[[photo.title]]&lt;/span>
            &lt;/div>
            &lt;div class="text">Some text goes here&lt;/div>
        &lt;/div>
    &lt;/template>
    &lt;script>
        Polymer({
            is: 'image-card'
        });
    &lt;/script>
&lt;/dom-module>

&lt;dom-module id="card-list">
  &lt;template>
    &lt;style>
      :host {

      }

      .cards {
          @apply(--layout-horizontal);
          @apply(--layout-center-justified);
          @apply(--layout-wrap);
      }

      image-card {
          margin: 10px;
      }
    &lt;/style>

    &lt;iron-ajax id="ajax" loading="{{loadingPhotos}}" url="[[_getAPIEndpoint(apiKey, searchText, page)]]"
               handle-as="text" on-response="_didReceiveResponse">&lt;/iron-ajax>

    &lt;div class="cards">
        &lt;template is="dom-repeat" items="[[photos]]">
            &lt;image-card class="card" photo="[[item]]">&lt;/image-card>
        &lt;/template>
    &lt;/div>
    &lt;/template>
    &lt;script>
    Polymer({
        is: 'card-list',

        behaviors: [
            Polymer.NeonAnimationRunnerBehavior
        ],

        properties: {
            apiKey: {
                type: String,
                value: 'c304f1096a06486d3c1e7ab271bf7f3f'
            },
            photos: Array,
            perPage: {
                type: Number,
                value: 4
            },
            page: {
                type: Number,
                value: 0
            },
            searchText: {
                type: String,
                value: 'Big Sur'
            },
            loadingPhotos: Boolean,

            animationConfig: {
                type: Object,
                value: function () {
                    return {
                        'entry': [{
                            name: 'cascaded-animation',
                            animation: 'transform-animation',
                            transformFrom: 'translateY(100%)',
                            transformTo: 'none',
                            timing: {
                                delay: 50
                            }
                        }],

                        'spin': [{
                            name: 'scale-up-animation'
                        }]
                    }
                }
            }
        },

        _getAPIEndpoint: function (apiKey, searchText, page) {
            return 'https://api.flickr.com/services/rest/?method=flickr.photos.search' +
                '&api_key=' + apiKey +
                '&safe_search=1&sort=interestingness-desc' +
                '&text=' + encodeURIComponent(searchText) +
                '&page=' + page +
                '&format=json' +
                '&per_page=' + this.perPage;
        },

        ready: function () {
            this._loadMorePhotos();
        },

        _loadMorePhotos: function () {
            if (this.$.ajax.lastRequest) {
                this.$.ajax.lastRequest.abort();
            }
            this.page++;
            this.$.ajax.generateRequest();
        },

        _didReceiveResponse: function (e) {
            var payload = JSON.parse(e.detail.response.match('jsonFlickrApi\\((.*)\\)')[1]);

            if (!payload || !payload.photos || !payload.photos.photo) {
                return;
            }

            this.photos = payload.photos.photo;
            this.async(function () {
                var nodeList = Polymer.dom(this.root).querySelectorAll('.card');
                this.animationConfig['entry'][0].nodes = Array.prototype.slice.call(nodeList);
                this.playAnimation('entry');
            });

            if (this._spinTimer) {
                clearTimeout(this._spinTimer);
                delete this._spinTimer;
            }
            this._spinIndex = 0;
            this._spinTimer = setInterval(this.spinAnimation.bind(this), 2000);
        },


        _spinTimer: null,
        _spinIndex: 0,
        spinAnimation: function () {
            this.async(function () {
                if (this._spinIndex >= this.perPage)
                    this._spinIndex = 1;
                else
                    this._spinIndex++;
                var node = Polymer.dom(this.root).querySelector('.card:nth-of-type(' + this.perPage + 'n+' + this._spinIndex + ')');
                if (node) {
                    this.animationConfig['spin'][0].node = node;
                    this.playAnimation('spin');
                }
            });
        }
    });
    &lt;/script>
&lt;/dom-module>  
&lt;/head>
    &lt;body class="fullbleed" unresolved>
        &lt;main-page>&lt;/main-page>
    &lt;/body>
&lt;/html></code></pre>

**Comment by govis** [comment]
I might be closing in on the actual issue and it seems to be related to running multiple animations sequentially. Still not sure how to resolve it, tried calling <code>cancelAnimation</code>, switching from <code>async</code> to <code>requestAnimationFrame</code>.

Is there a way to run some sort of a clean-up routine to reset the state of the animation?
